codegen/smithy-ruby-codegen-test/integration-specs/interceptor_spec.rb

```diff
@@ -0,0 +1,27 @@
+# frozen_string_literal: true
+
+require_relative 'spec_helper'
+
+module WhiteLabel
+  describe Client do
+    let(:test_interceptor) do
```

Comment:
This creates an anonymous class right? I'm not sure what rspec and rubocop recommend - I thought it complains about this, but I've been doing it like this:

```
let(:interceptor_class) do
  class TestInterceptor
    # normal class stuff
  end
end
```

and then creating with `interceptor_class.new`

Reply:
I believe anonymous classes are a preferred style.  See: https://github.com/rubocop/rspec-style-guide#declare-constants.

I think the code you had may leak the `TestInterceptor` name and should instead be:

```
let(:interceptor_class) do
  Class.new do
    # normal class stuff
  end
end
```

Reply:
Ah ok. I think `Class.new('TestInterceptor')` makes sense, then interceptor_class.new in the test. I should see if we have other examples in Hearth that should probably be that way.

Reply:
I think you don't want the name pass to `Class.new` - Thats usually used for extending a class.  Eg: `Class.new(ActiveRecord)` to extend ActiveRecord.

Reply:
Oh, I see. I thought that was for the class name. You're right.

Reply:
I do prefer `let()` always defining one thing (a class) and not a class and then an instance of it.